core/src/main/java/com/tweet/core/model/User.java

```diff
@@ -1,81 +1,92 @@
+    @Column(name = "user_id", nullable = false)
     private Long id;
+
+    @Column(name = "username", nullable = false)
     private String username;
+
+    @Length(min = 8)
+    @Column(name = "password", nullable = false)
+    private String password;
+
+    @Email
+    @Column(name = "email", nullable = false)
     private String email;
+
+    @Column(name = "first_name")
     private String firstName;
+
+    @Column(name = "surname")
     private String surname;
 
+    @Column(name = "active")
+    private Integer active;
+
     @ManyToMany(cascade = CascadeType.ALL)
-    @JoinTable(name = "role_mapping",
-            joinColumns = @JoinColumn(name = "user_id"),
-            inverseJoinColumns = @JoinColumn(name = "role_id"))
+    @JoinTable(name = "role_mapping", joinColumns = @JoinColumn(name = "user_id"), inverseJoinColumns = @JoinColumn(name = "role_id"))
     private Set<UserRole> userRoles;
 
-    @OneToMany(mappedBy = "user", targetEntity = Tweet.class)
+    @OneToMany(mappedBy = "userId")
     private Set<Tweet> tweets;
 
-    public Long getId() {
-        return id;
-    }
+    @OneToOne(mappedBy = "userId")
+    private Settings settings;
 
-    public void setId(Long id) {
-        this.id = id;
-    }
-
-    public String getUsername() {
-        return username;
-    }
-
-    public void setUsername(String username) {
-        this.username = username;
-    }
-
-    public String getEmail() {
-        return email;
-    }
-
-    public void setEmail(String email) {
-        this.email = email;
-    }
-
-    public String getFirstName() {
-        return firstName;
-    }
-
-    public void setFirstName(String firstName) {
-        this.firstName = firstName;
-    }
-
-    public String getSurname() {
-        return surname;
-    }
-
-    public void setSurname(String surname) {
-        this.surname = surname;
-    }
-
-    public Set<Tweet> getTweets() {
-        return tweets;
-    }
-
-    public void setTweets(Set<Tweet> tweets) {
-        this.tweets = tweets;
+    @ManyToMany(cascade = CascadeType.ALL)
+    @JoinTable(name = "followers_mapping",
+            joinColumns = {@JoinColumn(name = "follower_id")},
+            inverseJoinColumns = {@JoinColumn(name = "followed_id")})
+    private Set<User> following;
+
+    @JsonIgnore
+    @ManyToMany(mappedBy = "following")
+    private Set<User> followers;
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        User user = (User) o;
+        return Objects.equals(getId(), user.getId())
+                && Objects.equals(getFirstName(), user.getFirstName())
+                && Objects.equals(getSurname(), user.getSurname());
     }
 
-    public Set<UserRole> getUserRoles() {
-        return userRoles;
+    @Override
+    public int hashCode() {
+        return Objects.hash(getId(), getFirstName(), getSurname());
```

Comment:
We can’t use an auto-incrementing database id in the hashCode method since the transient and the attached object versions will no longer be located in the same hashed bucket.

Reply:
@NaturalId may be perfect choice for entity hashcode

for instance adding something like this:
@NaturalId
    private String isbn;
and than hashcode may be implemented as:
    @Override
    public int hashCode() {
        return Objects.hash(getIsbn());
    }

and tada! you have truly immutable hashcode even for very mutable entity.

NaturalId will generate id based java.util.UUID#randomUUID method. We still want to keep @Id as surrogate key generated by Hibernate.